packages/apidom-parser/src/parser.ts

```diff
@@ -18,119 +17,124 @@ export type {
-    this.findNamespace = async function findNamespace(
-      source: string,
-      options: ApiDOMParserOptions = {},
-    ) {
-      const adapter = await findAdapter(source, options.mediaType);
+  async findNamespace(
+    source: string,
+    options: ApiDOMParserOptions = {},
+  ): Promise<Namespace | undefined> {
+    const adapter = await this.findAdapter(source, options.mediaType);
 
-      return adapter?.namespace;
-    };
+    return adapter?.namespace;
+  }
 
-    this.findMediaType = async function findMediaType(source: string) {
-      const adapter = await findAdapter(source, undefined);
+  async findMediaType(source: string): Promise<string | void> {
+    const adapter = await this.findAdapter(source, undefined);
 
-      if (typeof adapter === 'undefined') {
-        return new MediaTypes().unknownMediaType;
-      }
+    if (typeof adapter === 'undefined') {
+      return new MediaTypes().unknownMediaType;
+    }
 
-      if (typeof adapter.mediaTypes === 'undefined') {
-        return new MediaTypes().unknownMediaType;
-      }
+    if (typeof adapter.mediaTypes === 'undefined') {
+      return new MediaTypes().unknownMediaType;
+    }
 
-      if (typeof adapter.detectionRegExp === 'undefined') {
-        return adapter.mediaTypes.latest();
-      }
+    if (typeof adapter.detectionRegExp === 'undefined') {
+      return adapter.mediaTypes.latest();
+    }
 
-      const { detectionRegExp } = adapter;
-      const matches = source.match(detectionRegExp);
+    const { detectionRegExp } = adapter;
+    const matches = source.match(detectionRegExp);
 
-      if (matches === null) {
-        return new MediaTypes().unknownMediaType;
-      }
+    if (matches === null) {
+      return new MediaTypes().unknownMediaType;
+    }
 
-      const { groups } = matches;
-      const version = groups?.version || groups?.version_json || groups?.version_yaml;
-      const format = groups?.version_json ? 'json' : groups?.version_yaml ? 'yaml' : 'generic';
+    const { groups } = matches;
+    const version = groups?.version || groups?.version_json || groups?.version_yaml;
+    const format = groups?.version_json ? 'json' : groups?.version_yaml ? 'yaml' : 'generic';
 
-      if (typeof version === 'undefined') {
-        return adapter.mediaTypes.latest();
-      }
+    if (typeof version === 'undefined') {
+      return adapter.mediaTypes.latest();
+    }
 
-      // @ts-ignore
-      return adapter.mediaTypes.findBy(version, format);
-    };
-
-    this.parse = async function parse(source: string, options: ApiDOMParserOptions = {}) {
-      let adapter;
-
-      try {
-        adapter = await findAdapter(source, options.mediaType);
-      } catch (error: unknown) {
-        throw new ParserError(
-          'Encountered an unexpected error while matching parser adapters against the source.',
-          {
-            source,
-            parserOptions: options,
-            cause: error,
-          },
-        );
-      }
+    // @ts-ignore
```

Comment:
IMHO we will be able to remove this already.
```suggestion
    // @ts-ignore
```